process/block/baseProcess.go

```diff
@@ -27,6 +27,24 @@ type hashAndHdr struct {
 	hash []byte
 }
 
+type nonceAndHashInfo struct {
+	hash  []byte
+	nonce uint64
+}
+
+type hdrInfo struct {
+	usedInBlock bool
+	hdr         data.HeaderHandler
+}
+
+type hdrForBlock struct {
```

Comment:
I think we might create a separate component for this, and have interaction through functions ? and doing it in a separate file

Reply:
Yes, we already discussed about this and I have created a task in Jira EN-4278 for refactor the management of requested/received of missed: txs, shard blocks, meta blocks into separate component. This is why I already extracted some structs related to this functionality in baseProcess